diff --git a/components/esp_matter/data_model/legacy/esp_matter_feature.cpp b/components/esp_matter/data_model/legacy/esp_matter_feature.cpp
index 7720c9b..7eeb0e5 100644
--- a/components/esp_matter/data_model/legacy/esp_matter_feature.cpp
+++ b/components/esp_matter/data_model/legacy/esp_matter_feature.cpp
@@ -2763,7 +2763,8 @@ esp_err_t add(cluster_t *cluster, config_t *config)
     uint32_t rid = feature::rfid_credential::get_id();
     uint32_t fgp = feature::finger_credentials::get_id();
     uint32_t face = feature::face_credentials::get_id();
+    uint32_t aliro = feature::aliro_provisioning::get_id();
     uint32_t feature = get_feature_map_value(cluster);
-    VerifyOrReturnError((feature & (pin | rid | fgp | face)) != 0, ESP_FAIL, ESP_LOGE(TAG, "Should add at least one of PIN, RID, FGP and FACE feature before add USR feature"));
+    VerifyOrReturnError((feature & (pin | rid | fgp | face | aliro)) != 0, ESP_FAIL, ESP_LOGE(TAG, "Should add at least one credential feature before adding the USR feature"));
-
+
     update_feature_map(cluster, get_id());
diff --git a/examples/door_lock/main/app_main.cpp b/examples/door_lock/main/app_main.cpp
index 21d0641..8f81079 100644
--- a/examples/door_lock/main/app_main.cpp
+++ b/examples/door_lock/main/app_main.cpp
@@ -176,17 +176,13 @@ extern "C" void app_main()
     door_lock_config.door_lock.delegate = &AliroDoorLockDelegate::Instance();
 #endif
     door_lock_config.door_lock.lock_state = chip::to_underlying(DoorLock::DlLockState::kLocked);
-    cluster::door_lock::feature::credential_over_the_air_access::config_t cota_config;
-    cluster::door_lock::feature::pin_credential::config_t pin_credential_config;
     cluster::door_lock::feature::user::config_t user_config;
     // endpoint handles can be used to add/modify clusters.
     endpoint_t *endpoint = door_lock::create(node, &door_lock_config, ENDPOINT_FLAG_NONE, NULL);
     ABORT_APP_ON_FAILURE(endpoint != nullptr, ESP_LOGE(TAG, "Failed to create door lock endpoint"));
     cluster_t *door_lock_cluster = cluster::get(endpoint, DoorLock::Id);
-    cluster::door_lock::feature::credential_over_the_air_access::add(door_lock_cluster, &cota_config);
-    cluster::door_lock::feature::pin_credential::add(door_lock_cluster, &pin_credential_config);
-    cluster::door_lock::feature::user::add(door_lock_cluster, &user_config);
 #ifdef CONFIG_ENABLE_ALIRO_OVER_NFC
     cluster::door_lock::feature::aliro_provisioning::add(door_lock_cluster);
 #endif
+    cluster::door_lock::feature::user::add(door_lock_cluster, &user_config);
     cluster::door_lock::attribute::create_auto_relock_time(door_lock_cluster, 5);
